File: Panoramix.cpp
#include <bits/stdc++.h>
using namespace std;

bool prime(int x) {
    if (x<=1)
        return false;
    else {
        for (int i=2; i<x; i++) {
            if (x%i==0)
                return false;
        }
        return true;
    } 
}

int main(void) {
    int n, m;
    cin >> n >> m;
    bool ans = false;
    for (int i=n+1; i<=m; i++) {
        if (prime(i)) {
            if (i==m) {
                ans = true;
                break;
            }
            else {
                ans = false;
                break;
            }
        }
        else {
            ans = false;
        }
    }
    if (ans)
        cout << "YES\n";
    else 
        cout << "NO\n";

    return 0;
}

File: Python/Panoramix.py

def prime(x):
    if x <= 1:
        return False
    else:
        for i in range(2, x):
            if x % i == 0:
                return False
        return True


def main():
    n = int(input())
    m = int(input())
    ans = False
    for i in range(n+1, m+1):
        if prime(i):
            if i == m:
                ans = True
            else:
                ans = False
                break
        else:
            ans = False
    if ans:
        print("YES")
    else:
        print("NO")


if __name__ == '__main__':
    main()
